mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessage.java

```diff
@@ -251,8 +251,7 @@ public final boolean equals(Object o) {
 
     @Override
     public final int hashCode() {
-        return Objects.hash(messageId, originMailboxes, owner, content, deliveryDate, deletionDate, sender, recipients,
-            subject, hasAttachment);
+        return Objects.hash(messageId, originMailboxes, owner, deliveryDate, deletionDate, sender, recipients, subject, hasAttachment);
```

Comment:
Well, writing the contract failed because equality between defferent kind of input streams is not handled by `Objects::equals`

I wrote a byte array equality stuff there but it ended up being not convincing either:
 - error management in equals and hashCode felt wrong
 - EqualsVerifier complained.

Then I went for the "less bad" option: testing content separately, like we do for every other classes.

Reply:
You can very well compute a hash of the stream at construction time (and enforce it's immutable). That way, you don't require to handle errors in hashCode/equals.
Otherwise, I suggest you don't provide equality has an object property but instead rely on an external Comparator/Equality implementation.